DetectorDescription/DDCMS/plugins/DDTestNavigateGeometry.cc

```diff
@@ -0,0 +1,97 @@
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DetectorDescription/DDCMS/interface/DetectorDescriptionRcd.h"
+#include "DetectorDescription/DDCMS/interface/DDDetector.h"
+#include "DD4hep/Detector.h"
+#include "DD4hep/DD4hepRootPersistency.h"
+#include "DD4hep/DetectorTools.h"
+#include "DD4hep/VolumeProcessor.h"
+
+#include <iostream>
+#include <string>
+
+using namespace std;
+using namespace cms;
+using namespace dd4hep;
+
+namespace {
+  
+  class VolumeProcessor : public dd4hep::PlacedVolumeProcessor {
+  public:
+    VolumeProcessor() = default;
+    ~VolumeProcessor() override = default;
+    
+    /// Callback to retrieve PlacedVolume information of an entire Placement
+    int process(dd4hep::PlacedVolume pv, int level, bool recursive) override {
+      m_volumes.emplace_back(pv.name());
+      int ret = dd4hep::PlacedVolumeProcessor::process(pv, level, recursive);
+      m_volumes.pop_back();
+      return ret;
+    }
+    
+    /// Volume callback
+    int operator()(dd4hep::PlacedVolume pv, int level) override {
+      dd4hep::Volume vol = pv.volume();
+      cout << "Hierarchical level:" << level << "   Placement:";
```

Comment:
@ianna I understand this is just a test class, but would not be better to use MessageLogger instead of cout? Or this is supposed to be never used in a production workflow and with multithreading enabled, in which case even cout could be ok?

Reply:
@fabiocos - yes, it's just for a test: as you can see it's in a plugin in an anonymous namespace.